docs/cookbook/dynamic-configuration/02_Templates/01_Looping.md

```diff
@@ -0,0 +1,42 @@
+# 01 Basics of Templating
+
+The templating is based on the excellent [gomplate](https://docs.gomplate.ca/) library. Currently using a custom fork to allow loading gomplate as a library in addition to some new commands. This will NOT try to cover the full range of gomplate, would recommend reading the documentation for full knowledge. 
+
+`docker run -v ${PWD}/:/etc/grafana grafana/agentctl:latest template-parse /etc/grafana/01_config.yml`
+
+## Looping
+
+[agent-1.yml](./01_assets/agent-1.yml)
+
+```yaml
+server:
+  http_listen_port: 12345
+  log_level: debug
+metrics:
+  wal_directory: /tmp/grafana-agent-normal
+  global:
+    scrape_interval: 60s
+    remote_write:
+      - url: https://prometheus-us-central1.grafana.net/api/prom/push
+        basic_auth:
+          username: 12345
+          password: secretpassword
+  configs:
+    - name: default
+  {{ range slice "apple" "banana" "pear" }}
+    - name: {{ . }}
+  {{ end }}
```

Comment:
WDYT about the current limitation that looping can only be done at the base config template level? Is that going to restrict any common use cases?

Reply:
I am unsure what you mean by base config template level. Can you elaborate?

Reply:
Ah, sorry, I mean that AFAICT you can't iterate inside of an individual template file i.e., `metrics-instance-*.yml`